app/cdap/components/hydrator/components/TopPanel/ActionButtons/index.tsx

```diff
@@ -0,0 +1,295 @@
+          <PreviewModeButton
+            variant="text"
+            onClick={togglePreviewMode}
+            data-cy="preview-active-btn"
+          >
+            <div>
+              <IconSVG name={'icon-eye'}></IconSVG>
+              <ButtonLabel>Preview</ButtonLabel>
+            </div>
+          </PreviewModeButton>
+          <CommonButton
+            active={viewConfig}
+            disabled={previewLoading || previewRunning}
+            variant="text"
+            onClick={!previewLoading && !previewRunning && toggleConfig}
+            data-cy="preview-config-btn"
+          >
+            <div>
+              <IconSliders name={'icon-sliders'}></IconSliders>
+              <ButtonLabel>Configure</ButtonLabel>
+            </div>
+          </CommonButton>
+          {!previewLoading &&
+            (!previewRunning ? (
+              <CustomTooltip
+                title={hasNodes ? '' : 'Start building a pipeline before starting preview'}
+                arrow
+                placement="bottom"
+              >
+                <span>
+                  <CommonButton
+                    disabled={!hasNodes}
+                    variant="text"
+                    onClick={hasNodes && startOrStopPreview}
+                    data-cy="preview-top-run-btn"
+                  >
+                    <div>
+                      <IconPlay name={'icon-play'} disabled={!hasNodes}></IconPlay>
+                      <ButtonLabel>Run</ButtonLabel>
+                    </div>
+                  </CommonButton>
+                </span>
+              </CustomTooltip>
+            ) : (
+              <CommonButton variant="text" onClick={startOrStopPreview} data-cy="stop-preview-btn">
+                <div>
+                  <IconStop name={'icon-stop'}></IconStop>
+                  <ButtonLabel>Stop</ButtonLabel>
+                </div>
+              </CommonButton>
+            ))}
+          {previewLoading && (
+            <CommonButton variant="text" data-cy="starting-preview-btn">
+              <div>
+                <IconLoading className="fa fa-refresh fa-spin"></IconLoading>
+                <ButtonLabel>{loadingLabel}</ButtonLabel>
+              </div>
+            </CommonButton>
+          )}
+          <CommonButton title={queueStatus} variant="text">
+            <div>
+              <RunTimeSpan>
+                {displayDuration.minutes}:{displayDuration.seconds}
+              </RunTimeSpan>
+              <ButtonLabel>{timerLabel}</ButtonLabel>
+            </div>
+          </CommonButton>
+          <CommonButton
+            disabled={!currentPreviewId}
+            active={viewLogs}
+            variant="text"
+            onClick={currentPreviewId ? onClickLogs : false}
+          >
+            <div>
+              <IconSVG name={'icon-file-text-o'}></IconSVG>
+              <ButtonLabel>Logs</ButtonLabel>
+            </div>
+          </CommonButton>
+        </div>
+      ) : (
+        <div>
```

Comment:
unsure if these divs are necessary for styling or grouping but feel free to use `<></>` (React.Fragment) more often.

Reply:
yeah they are more for grouping here, will update